### DISCUSSION TITLE
Getting jq error (at <stdin>:1): Cannot index string with string "tag_name" with simple API call

### DISCUSSION THREAD

**Discussion by akrherz** [community/community#33700] [Actions] [answered]
Greetings,

We have a [MacOS Github Action](https://github.com/igniterealtime/Openfire/blob/main/.github/workflows/macos-release-artifact.yml) that attempts to generate release artifacts.  We've hit a very strange error with a seemingly mundane request.  Our relevant yaml section is:

```yaml
      - id: get-id
        name: Compute needed variables
        run: |
          set -x
          id=$(echo ${{ github.repository }} | cut -d- -f2)
          echo "::set-output name=id::$id"
          echo "id is '$id'"
          tag=$(echo ${{ github.ref }} | cut -d '/' -f3)
          echo "::set-output name=tag::$tag"
          echo "tag is '$tag'"
          rel_id=$(curl -sL https://api.github.com/repos/${{github.repository}}/releases | jq -r --arg TAG "$tag" '.[] | select(.tag_name==$TAG) | .id')
          echo ::set-output name=rel_id::$rel_id
          echo "rel_id is '$rel_id'"
          version=$(echo ${{ github.ref }} | cut -d '/' -f3 | cut -c 2- | sed 's/\./_/g')
          echo "::set-output name=version::$version"
          echo "version is '$version'"
```

An example [CI failure](https://github.com/igniterealtime/Openfire/actions/runs/3098117393/jobs/5015619083) reports:

```
++ curl -sL https://api.github.com/repos/igniterealtime/Openfire/releases
++ jq -r --arg TAG 2113 '.[] | select(.tag_name==$TAG) | .id'
jq: error (at <stdin>:1): Cannot index string with string "tag_name"
```

This failure seems flakey / transient as yesterday we simply reordered the run commands and got a working result. Anybody have ideas on what is causing this to happen?

**Comment by akrherz** [comment]
We continue to see this issue: seemingly transient failures on the /releases API endpoint from GHA.

**Comment by airtower-luna** [comment, marked as answer]
Could it be that your request is getting rate limited or something? Adding `-v` to the `curl` command line and `echo`ing the returned data could show that. Or you could go right to authorizing the request with the `GITHUB_TOKEN`, and if that gets rid of the problem it probably was that.
